Core/MigrationService.php

```diff
@@ -437,7 +440,22 @@ protected function executeMigrationInner(Migration $migration, MigrationDefiniti
                 $currentUser = $this->getCurrentUser();
                 $this->authenticateUserByLogin($adminLogin ?? self::ADMIN_USER_LOGIN);
 
-                $this->repository->commit();
+                try {
+                    $this->repository->commit();
+                } catch (Exception $e) {
+                    if ($executor && !$executor instanceof SQLExecutor) {
+                        throw $e;
+                    }
+                    // when executing sql command, we might get an exception in the case where transaction has been auto-commited by driver
+                    // (when using drop table or create table)
+                    $previous = $e->getPrevious();
+                    $possiblePdoException = $previous ? $previous->getPrevious() : null;
+                    if (!($possiblePdoException instanceof PDOException && $possiblePdoException->getMessage() === 'There is no active transaction')) {
+                        throw $e;
+                    }
+
+                    $executor->resetTransactionDepth();
```

Comment:
I wonder if this is necessary... does DBAL throw if it finds at end of php script that there are some supposedly pending transactions? If not, it might not be worth the hassle. After all, there might be other cases where a ddl gets executed, and the dbal's transactionNestingLevel gets out of sync

Reply:
It seems that this is necessary after all. At least for the test suite, which can execute many migrations one after another, within the same process. But the same applies to the `migrate` command